stock_split_picking/wizards/stock_picking_split.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013-2018 Camptocamp SA - Alexandre Fayolle
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import _, api, models, fields
+from odoo.exceptions import ValidationError
+from odoo.addons import decimal_precision as dp
+from odoo.tools import float_compare, float_is_zero
+
+
+class StockPickingSplit(models.TransientModel):
+    _name = 'stock.picking.split'
+
+    picking_id = fields.Many2one(
+        'stock.picking',
+        string='Picking',
+        required=True,
+        default=lambda s: s.env.context.get('active_id', False)
+    )
+    line_ids = fields.One2many(
+        'stock.picking.split.line',
+        'split_id',
+        string='Split Operations',
+    )
+
+    @api.model
+    def create(self, vals):
+        rec = super(StockPickingSplit, self).create(vals)
+        rec._generate_lines()
+        return rec
+
+    @api.multi
+    def process(self):
+        self.ensure_one()
+        uom_dp = self.env['decimal.precision'].precision_get(
+            'Product Unit of Measure')
+        picking = self.picking_id
+        reserve = False
+        if picking.pack_operation_ids:
+            reserve = True
+            picking.do_unreserve()
+        split_picking = picking.copy(
+            {
+                'name': '/',
+                'move_lines': [],
+                'pack_operation_ids': [],
+            }
+        )
+        # Handle the case where there are several stock.move with the same
+        # product+UoM. We summed the qtys for all these moves in the wizard,
+        # and now we need to spread back the qtys among the different moves
+        # with the same product+UoM.
+        product_split = {}
+        for line in self.line_ids:
+            key = (line.product_id, line.product_uom_id, line.full_split)
+            product_split[key] = line.split_qty
+        new_move_ids = []
+        for move in picking.move_lines:
```

Comment:
before wizard creation we union lines per product and uom, and here we also should use same approach 
we need exclude lines with same product and uom if they more that on (otherwise you will have lines with 0 qty)

Reply:
You mean we can get 2 moves corresponding with the same key? I think it is handled in the lines after. In the initial PR there is a comment from @gurneyalex explaining this: https://github.com/OCA/stock-logistics-workflow/pull/471#discussion_r234927811

Reply:
yes, I saw it and works while you have simple product, but when you track it by lots error raised or from  181 or 193 line
I think maybe we not divide them but split one by one, for instance if you have 2 lines with same product per 10 unit each, and split them in 7 and 13 one pick will have one line with 7 unit and other with 10 and 3, in this case it will be easier to send lots or s/n if we know them
and while written it,  realized we can't split product which tracking by sn on float number, only integers allowed

Reply:
I tested and it is working "well", we are losing the lots/serial number which were already assigned (pack operation removed by the unreserve step), and the reserved ones in the new picking may be different. I don't know if it is an issue.

Good catch about the fact we should not be able to split a move with floating numbers if its product is tracked by serial number, I'll add a check on this.